lib/rules/font-family-name-quotes/index.mjs

```diff
@@ -187,75 +181,57 @@ const rule = (primary, _secondary, context) => {
 			switch (primary) {
 				case 'always-unless-keyword':
 					if (!hasQuotes) {
-						if (context.fix) {
-							fontFamilyNode.addQuotes();
-
-							return;
-						}
-
-						return complain(messages.expected(family), rawFamily, decl);
+						return complain('expected', fontFamilyNode, decl);
 					}
 
 					return;
 
 				case 'always-where-recommended':
 					if (!recommended && hasQuotes) {
-						if (context.fix) {
-							fontFamilyNode.removeQuotes();
-
-							return;
-						}
-
-						return complain(messages.rejected(family), rawFamily, decl);
+						return complain('rejected', fontFamilyNode, decl);
 					}
 
 					if (recommended && !hasQuotes) {
-						if (context.fix) {
-							fontFamilyNode.addQuotes();
-
-							return;
-						}
-
-						return complain(messages.expected(family), rawFamily, decl);
+						return complain('expected', fontFamilyNode, decl);
 					}
 
 					return;
 
 				case 'always-where-required':
 					if (!required && hasQuotes) {
-						if (context.fix) {
-							fontFamilyNode.removeQuotes();
-
-							return;
-						}
-
-						return complain(messages.rejected(family), rawFamily, decl);
+						return complain('rejected', fontFamilyNode, decl);
 					}
 
 					if (required && !hasQuotes) {
-						if (context.fix) {
-							fontFamilyNode.addQuotes();
-
-							return;
-						}
-
-						return complain(messages.expected(family), rawFamily, decl);
+						return complain('expected', fontFamilyNode, decl);
 					}
 			}
 		}
 
 		/**
-		 * @param {string} message
-		 * @param {string} family
+		 * @param {'expected'|'rejected'} messageType
+		 * @param {MutableNode} fontFamilyNode
 		 * @param {import('postcss').Declaration} decl
 		 */
-		function complain(message, family, decl) {
+		function complain(messageType, fontFamilyNode, decl) {
+			const { name, rawName } = fontFamilyNode;
+			const fix = () => {
+				const mutate = messageType === 'expected' ? 'addQuotes' : 'removeQuotes';
+				const end = rawName.length - 1;
+				const word = messageType === 'expected' ? `"${rawName}"` : rawName.slice(1, end);
+
+				fontFamilyNode[mutate]();
+
+				return decl.rangeBy({ word });
+			};
```

Comment:
[suggest] What about an idea that `addQuotes` and `removeQuotes` return a range since they already have the `slice` logic? For example:

```js
{
	removeQuotes() {
		// ...
		return decl.rangeBy({ word: this.name });
	},
	addQuotes() {
		// ...
		return decl.rangeBy({ word: fixedName });
	},
}
```

Then, `fix` would be much simpler:

```js
const fix = () => {
	return messageType === 'expected'
		? fontFamilyNode.addQuotes()
		: fontFamilyNode.removeQuotes();
	};
```

Reply:
`no-confusing-arrow` rule conflicts with prettier; it's kinda annoying.

Reply:
Agree. I noticed the [`no-confusing-arrow`](https://eslint.org/docs/latest/rules/no-confusing-arrow) rule had been deprecated because of the stylistic rule.

I will remove it from our shared config:
https://github.com/stylelint/eslint-config-stylelint/blob/c3d49f9cd9040aee87132b2b070b4685be963703/index.js#L26-L31

Reply:
Opened https://github.com/stylelint/eslint-config-stylelint/pull/276